len8/cli.py

```diff
@@ -26,42 +26,74 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-import subprocess as sp
 import sys
+import typing as t
 from pathlib import Path
 
-from len8.models.parser import Parser
+import click
 
+from len8 import Checker
+from len8.errors import BadLines, InvalidPath
 
-def run(command: str):
-    if sys.version_info >= (3, 7, 0):
-        return sp.run(command, shell=True, capture_output=True)
 
-    if os.name != "nt":
-        return sp.run(f"{command} > /dev/null 2>&1", shell=True)
+def _as_paths(value: str) -> t.Tuple[Path, ...]:
+    if not value:
+        return ()
 
-    # Windows users will have to put up with the output for 3.6 tests.
-    return sp.run(command, shell=True)
+    return tuple(Path(p) for p in value.split(","))
 
 
-def test_basic_parsing() -> None:
-    sys.argv = ["len8", "."]
-    parser = Parser()
-    assert parser.paths == [Path(".")]
-    assert parser.exclude == []
-    assert parser.extend == False
+@click.command()
+@click.version_option()
+@click.argument("paths", type=Path, required=True, nargs=-1)
+@click.option(
+    "-x",
+    "--exclude",
+    type=_as_paths,
+    default="",
+    metavar="FILEPATH",
+    help="Comma-separated list of files/dirs to exclude.",
+)
+@click.option(
+    "-l",
+    "--extend-length",
+    count=True,
+    help=(
+        "Increase line length for code. Option is additive, and can be used "
+        "up to 2 times (corresponding to 88 and 99; omit for 79)"
+    ),
+)
+@click.option(
+    "-c",
+    "--code-length",
+    type=int,
+    metavar="CHARS",
+    help="Custom line length for code. Overrides --length if set.",
```

Comment:
`--length` is not a valid option, either change it above to `-l`, `--length`, or change it here to `--extend-length`